### PR TITLE
[Compatibility] Added GETSET, SETNX, SUBSTR deprecated commands by mapping to existing commends

### PR DESCRIPTION
**Pull request by Vijay-Nirmal**
- [x] Add GETSET, SETNX, SUBSTR option for HSCAN
- [x] Updated Test cases
- [x] Updated documentation for GETSET, SETNX, SUBSTR
- [x] Marked all deprecated comments in the doc

Part 1 of #684

### CODE REVIEW

**Review by vazois — CHANGES_REQUESTED** [review]
Can you please add a slot verification test in ClusterSlotVeficationTests?

**Comment by Vijay-Nirmal** [comment]
@vazois Added ClusterSlotVeficationTests

File: libs/server/Resp/BasicCommands.cs [hunk 1]
```diff
@@ -267,6 +267,22 @@ private bool NetworkSET<TGarnetApi>(ref TGarnetApi storageApi)
...
+        /// GETSET
+        /// </summary>
+        private bool NetworkGETSET<TGarnetApi>(ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            Debug.Assert(parseState.Count == 2);
+            Debug.Assert(parseStateBuffer.Length >= 3);
+            var key = parseState.GetArgSliceByRef(0);
+            var value = parseState.GetArgSliceByRef(1);
+            var getOption = ArgSlice.FromPinnedSpan([(byte)'G', (byte)'E', (byte)'T']);
```
[the remark this answers is not in the document]
Comment (vazois) [reply libs/server/Resp/BasicCommands.cs:280]: You can add it in CmdStrings.cs

**Review by vazois — CHANGES_REQUESTED** [review]

File: libs/server/Resp/BasicCommands.cs [hunk 2]
```diff
@@ -390,6 +406,22 @@ private bool NetworkSETEX<TGarnetApi>(bool highPrecision, ref TGarnetApi storage
...
+        /// SETNX
+        /// </summary>
+        private bool NetworkSETNX<TGarnetApi>(bool highPrecision, ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            Debug.Assert(parseState.Count == 2);
+            Debug.Assert(parseStateBuffer.Length >= 3);
+            var key = parseState.GetArgSliceByRef(0);
+            var value = parseState.GetArgSliceByRef(1);
+            var getOption = ArgSlice.FromPinnedSpan([(byte)'N', (byte)'X']);
```
[the remark this answers is not in the document]
Comment (Vijay-Nirmal) [reply libs/server/Resp/BasicCommands.cs]: Fixed

File: libs/server/Resp/BasicCommands.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vijay-Nirmal) [reply libs/server/Resp/BasicCommands.cs]: Fixed

**Review by vazois — APPROVED** [review]

**Comment by vazois** [comment]
Please resolve conflicts

**Pull request merged**